### DISCUSSION TITLE
Questions on sliding window analysis.

### DISCUSSION THREAD

**Discussion by andrewcz** [jakobrunge/tigramite#254] [Q&A] [answered]
**In summary i am trying to answer the following questions -
Is there a lag consistent/persistent through time
What is the strength of lag
What is time delay of the lag**

So my thought is that the sliding window analysis is for understanding persistence in causal effects.

My question is what is the difference between - 

window_step=1
window_length=7

If I have a set of financial time series described by the below code. I would think that the lead lag happens quite quickly hence the short window step (?). the below is my starting code 

```
import pandas as pd
import yfinance as yf


df = yf.download(['AAPL','GOOG','MSFT','DIS', '^GSPC'], start="2021-01-01")['Adj Close'].dropna()
dataframe=df.pct_change().dropna()
data=dataframe.to_numpy()
data.shape
var_names = ['AAPL','GOOG','MSFT','DIS', '^GSPC']
dataframe = pp.DataFrame(data, var_names=var_names)
tp.plot_timeseries(dataframe); plt.show()

N = 5
T = data.shape[0]
datatime = np.arange(T)
```

**Comment by andrewcz** [comment]
The visual get from running the code - 

![image](https://user-images.githubusercontent.com/2296335/200147242-07eb6df6-cc81-416d-88d2-6406f2062b2d.png)


Please assist with comments or suggestions @jakobrunge

**Comment by jakobrunge** [comment, marked as answer]
window_step is the stepsize in units of the time series data resolution of the windows over which each causal graph is estimated.

**Reply by jakobrunge** [reply]
window_step and window_size have not much to do with time-lags, it's just that tau_max should be much smaller than window_size, otherwise you have no samples left to estimate anything on.
